### PR TITLE
Docs - first attempt

### PR DESCRIPTION
**Pull request by keunwoochoi**
* need to configure `index.rst` etc better but it kinda works
* following google docstring format

### CODE REVIEW

**Comment by rabitt** [comment]
sorry for the newly merged PR... creating lots of conflicts with this one...

File: mirdata/beatles.py [hunk 1]
```diff
@@ -87,23 +127,59 @@ def download(data_home=None, force_overwrite=False):
...
+        (bool): True if Beatles dataset folder exists
+
+    """
     save_path = utils.get_save_path(data_home)
     dataset_path = os.path.join(save_path, DATASET_DIR)
     return os.path.exists(dataset_path)
 
 
 def validate(dataset_path, data_home=None):
+    """Validate if the stored dataset is a valid version
```
Comment (rabitt) [inline mirdata/beatles.py]: Validate if a local version of this dataset is consistent

File: mirdata/beatles.py [hunk 2]
```diff
@@ -87,23 +127,59 @@ def download(data_home=None, force_overwrite=False):
...
+
+    Args:
+        dataset_path (str): Beatles dataset local path
+        data_home (str): Local home path that the dataset is being stored
```
Comment (rabitt) [inline mirdata/beatles.py]: path that --> path where

File: mirdata/beatles.py [hunk 3]
```diff
@@ -87,23 +127,59 @@ def download(data_home=None, force_overwrite=False):
...
+
+    Returns:
+        missing_files (list): List of file paths that are in the dataset index
+            but missing locally
+        invalid_checksums (list): List of file paths that file exists in the dataset
+            index but has a different checksum compare to the reference checksum
```
Comment (rabitt) [inline mirdata/beatles.py]: "List of file paths where the expected file exists locally but has a different checksum than the reference"

File: mirdata/beatles.py [hunk 4]
```diff
@@ -87,23 +127,59 @@ def download(data_home=None, force_overwrite=False):
...
+
+    """
     missing_files, invalid_checksums = utils.validator(
         INDEX, data_home, dataset_path
     )
     return missing_files, invalid_checksums
 
 
 def track_ids():
+    """Return track ids
```
Comment (rabitt) [inline mirdata/beatles.py]: General comment, I don't like when the summary docs start with "return", isn't that what "Returns:" is for? :)

Instead, maybe "List of track IDs for this dataset"

File: mirdata/beatles.py [hunk 5]
```diff
@@ -87,23 +127,59 @@ def download(data_home=None, force_overwrite=False):
...
+        (list): A list of track ids
+    """
     return list(INDEX.keys())
 
 
 def load(data_home=None):
+    """Load Beatles dataset
+
+    Args:
+        data_home (str): Local home path that the dataset is being stored.
```
Comment (rabitt) [inline mirdata/beatles.py]: Location on disk where the dataset is stored. If `None`, looks for the data in the default directory, `~/mir_datasets`.

File: mirdata/beatles.py [hunk 6]
```diff
@@ -115,6 +191,12 @@ def load(data_home=None):
 
 
 def _load_beats(beats_path):
+    """Private function to load beat data for a track
+
+    Args:
+        beats_path (str):
+
```
Comment (rabitt) [inline mirdata/beatles.py]: Add Returns:

Comment (rabitt) [reply mirdata/beatles.py]: same comment for private loaders below

File: mirdata/beatles.py [hunk 7]
```diff
@@ -115,6 +191,12 @@ def load(data_home=None):
 
 
 def _load_beats(beats_path):
+    """Private function to load beat data for a track
```
Comment (rabitt) [inline mirdata/beatles.py]: Private function to load Beatles format beat data from a file.

(Similar comment for loaders below)

File: mirdata/beatles.py [hunk 8]
```diff
@@ -193,6 +294,8 @@ def _load_sections(sections_path):
 
 
 def _fix_newpoint(beat_positions):
+    """(placeholder)
```
Comment (rabitt) [inline mirdata/beatles.py]: Fills in missing beat position labels by inferring the beat position from neighboring beats.

File: mirdata/ikala.py [hunk 9]
```diff
@@ -1,11 +1,35 @@
 # -*- coding: utf-8 -*-
+"""iKala Dataset Loader
+
+The iKala dataset comprises of 252 30-second excerpts sampled from 206 iKala
```
Comment (rabitt) [inline mirdata/ikala.py]: comprises of --> is comprised of

File: mirdata/beatles.py [hunk 10]
```diff
@@ -16,6 +32,22 @@
 
 
 class Track(object):
+    """Beatles track class
+
+    Args:
+        track_id (str): track id of the track
+        data_home (str): data home folder path
```
Comment (lostanlen) [inline mirdata/beatles.py]: four juxtaposed nouns? what do they mean?

File: mirdata/beatles.py (hunk 6 above)
Comment (keunwoochoi) [reply mirdata/beatles.py]: Let me work on the private in the following PR.

File: mirdata/beatles.py (hunk 10 above)
Comment (keunwoochoi) [reply mirdata/beatles.py]: i was being german but @rabitt reviewed it above, i'm changing it now.

File: mirdata/beatles.py [hunk 11]
```diff
@@ -16,6 +32,22 @@
...
+
+    Attributes:
+        track_id (str): track id
+        audio_path (str): track audio path
+        title (str): title of the track
+        beats: beats beat annotation
```
[the remark this answers is not in the document]
Comment (keunwoochoi) [reply mirdata/beatles.py]: Where are these classes (`BeatData` for example) defined?

File: mirdata/beatles.py (hunk 4 above)
Comment (keunwoochoi) [reply mirdata/beatles.py]: yeah, well, or `get the list of..`?

**Pull request merged**